test/integration.js

```diff
@@ -740,5 +752,99 @@ tape('targetingRules: multiple rules are applied, pricingBounds are honored', as
 	t.end()
 })
 
+tape('validatorWorker: does not apply empty aggr', async function(t) {
+	const channel = getValidEthChannel()
+	channel.spec = {
+		...channel.spec,
+		targetingRules: []
+	}
+
+	// Submit a new channel; we submit it to both sentries to avoid 404 when propagating messages
+	await Promise.all([
+		fetchPost(`${leaderUrl}/channel`, dummyVals.auth.leader, channel),
+		fetchPost(`${followerUrl}/channel`, dummyVals.auth.follower, channel)
+	])
+
+	// update targetingRules
+	await fetchPost(`${leaderUrl}/channel/${channel.id}/events`, dummyVals.auth.creator, {
+		events: [
+			{
+				type: constants.eventTypes.update_targeting,
+				targetingRules: [
+					{ if: [{ eq: [{ get: 'country' }, 'US'] }, { set: ['price.IMPRESSION', { bn: '30' }] }] }
+				]
+			}
+		]
+	})
+
+	// sleep to give some time for the channel to be updated
+	await wait(cfg.CHANNEL_REFRESH_INTERVAL)
+
+	// wait for the events to be aggregated and new states to be issued
+	await aggrAndTick()
+	await forceTick()
+
+	const sentry = new SentryInterface(dummyAdapter, channel, { logging: false })
+	const data = await sentry.getLatestMsg(dummyVals.ids.leader, 'Accounting')
+
+	t.equal(data, null, 'shoud not produce new Accounting aggregate with empty event aggregate')
+
+	t.end()
+})
+
+tape('/validator-messages: reject Accounting/NewState messages with empty balances', async function(
+	t
```

Comment:
And here we test that the sentry never accepts` newstate/accounting` messages with empty balances